This file is part of MXE.
See index.html for further information.

Disable building and installing dynamic libraries in the CMakeLists.txt file.
(The file unfortunately does not provide switches for this purpose.)
diff -uNr flann-1.7.1-src/src/cpp/CMakeLists.txt flann-1.7.1-src.patched/src/cpp/CMakeLists.txt
--- flann-1.7.1-src/src/cpp/CMakeLists.txt	2011-12-23 12:16:09.000000000 +0100
+++ flann-1.7.1-src.patched/src/cpp/CMakeLists.txt	2012-10-01 12:36:59.013191056 +0200
@@ -24,29 +24,29 @@
     set_property(TARGET flann_cuda_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-    add_library(flann_cpp SHARED "")
-    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
-    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
-
-    if (BUILD_CUDA_LIB)
-	    cuda_add_library(flann_cuda SHARED "")
-        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
-        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
-    # 	target_link_libraries(flann_cuda cudpp_x86_64)
-    endif()
-else()
-    add_library(flann_cpp SHARED ${CPP_SOURCES})
-    if (BUILD_CUDA_LIB)
-		cuda_add_library(flann_cuda SHARED ${CPP_SOURCES})
-    endif()
-endif()
-
-set_target_properties(flann_cpp PROPERTIES
-   VERSION ${FLANN_VERSION}
-   SOVERSION ${FLANN_SOVERSION}
-   DEFINE_SYMBOL FLANN_EXPORTS
-) 
+#if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
+#    add_library(flann_cpp SHARED "")
+#    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
+#    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
+#
+#    if (BUILD_CUDA_LIB)
+#	    cuda_add_library(flann_cuda SHARED "")
+#        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
+#        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
+#    # 	target_link_libraries(flann_cuda cudpp_x86_64)
+#    endif()
+#else()
+#    add_library(flann_cpp SHARED ${CPP_SOURCES})
+#    if (BUILD_CUDA_LIB)
+#		cuda_add_library(flann_cuda SHARED ${CPP_SOURCES})
+#    endif()
+#endif()
+#
+#set_target_properties(flann_cpp PROPERTIES
+#   VERSION ${FLANN_VERSION}
+#   SOVERSION ${FLANN_SOVERSION}
+#   DEFINE_SYMBOL FLANN_EXPORTS
+#) 
 
 if (BUILD_CUDA_LIB)
     set_target_properties(flann_cuda PROPERTIES
@@ -57,12 +57,12 @@
 endif()
 
 
-#debug libraries
-add_library(flann_cpp-gd SHARED ${CPP_SOURCES})
-set_target_properties(flann_cpp-gd PROPERTIES 
-    COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG}
-    DEFINE_SYMBOL FLANN_EXPORTS
-)
+##debug libraries
+#add_library(flann_cpp-gd SHARED ${CPP_SOURCES})
+#set_target_properties(flann_cpp-gd PROPERTIES 
+#    COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG}
+#    DEFINE_SYMBOL FLANN_EXPORTS
+#)
 
 add_library(flann_cpp_s-gd STATIC ${CPP_SOURCES})
 set_target_properties(flann_cpp_s-gd PROPERTIES COMPILE_FLAGS ${CMAKE_CXX_FLAGS_DEBUG})
@@ -108,34 +108,35 @@
     endif()
     set_property(TARGET flann_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 
-    if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-        add_library(flann SHARED "")
-        set_target_properties(flann PROPERTIES LINKER_LANGUAGE CXX)
-        target_link_libraries(flann -Wl,-whole-archive flann_s -Wl,-no-whole-archive)
-    else()
-        add_library(flann SHARED ${C_SOURCES})
-    endif()
-
-    set_target_properties(flann PROPERTIES
-       VERSION ${FLANN_VERSION}
-       SOVERSION ${FLANN_SOVERSION}
-       DEFINE_SYMBOL FLANN_EXPORTS
-    ) 
+#    if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
+#        add_library(flann SHARED "")
+#        set_target_properties(flann PROPERTIES LINKER_LANGUAGE CXX)
+#        target_link_libraries(flann -Wl,-whole-archive flann_s -Wl,-no-whole-archive)
+#    else()
+#        add_library(flann SHARED ${C_SOURCES})
+#    endif()
+#
+#    set_target_properties(flann PROPERTIES
+#       VERSION ${FLANN_VERSION}
+#       SOVERSION ${FLANN_SOVERSION}
+#       DEFINE_SYMBOL FLANN_EXPORTS
+#    ) 
 endif()
 
 
 if(WIN32)
 if (BUILD_C_BINDINGS)
-    install (
-        TARGETS flann
-        RUNTIME DESTINATION share/flann/matlab
-    )
+#    install (
+#        TARGETS flann
+#        RUNTIME DESTINATION share/flann/matlab
+#    )
 endif()
 endif(WIN32)
 
 
 install (
-    TARGETS flann_cpp flann_cpp_s flann_cpp-gd flann_cpp_s-gd
+#    TARGETS flann_cpp flann_cpp_s flann_cpp-gd flann_cpp_s-gd
+    TARGETS flann_cpp_s flann_cpp_s-gd
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION ${FLANN_LIB_INSTALL_DIR}
     ARCHIVE DESTINATION ${FLANN_LIB_INSTALL_DIR}
@@ -152,7 +153,8 @@
 
 if (BUILD_C_BINDINGS)
     install (
-        TARGETS flann flann_s
+#        TARGETS flann flann_s
+        TARGETS flann_s
         RUNTIME DESTINATION bin
         LIBRARY DESTINATION ${FLANN_LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${FLANN_LIB_INSTALL_DIR}
